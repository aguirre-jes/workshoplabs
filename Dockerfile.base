# Golden Image Corporativa - Imagen Base Estandarizada
# Esta imagen sirve como punto de partida para todas las aplicaciones de la organización

# Partimos de una imagen pública, oficial y ligera como Alpine
FROM alpine:3.20

# Etiquetamos la imagen para identificar al responsable y la versión
LABEL maintainer="equipo.devops@miempresa.com"
LABEL version="1.0.0"
LABEL description="Golden Image corporativa basada en Alpine Linux"
LABEL created="2025-08-07"

# Instalamos nuestras herramientas base estándar y actualizamos los certificados
RUN apk add --no-cache \
    # Herramientas básicas del sistema
    bash \
    curl \
    wget \
    jq \
    git \
    # Certificados y seguridad
    ca-certificates \
    openssl \
    # Utilidades de tiempo y zona horaria
    tzdata \
    # Python runtime (común en la organización)
    python3 \
    py3-pip \
    # Herramientas de desarrollo comunes
    build-base \
    && update-ca-certificates \
    # Limpieza de cache
    && rm -rf /var/cache/apk/*

# Configuramos zona horaria corporativa
ENV TZ=America/Santo_Domingo
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone

# Creamos usuario estándar para aplicaciones (principio de mínimo privilegio)
RUN addgroup -g 1001 -S appgroup && \
    adduser -D -u 1001 -G appgroup -S appuser

# Configuramos variables de entorno corporativas
ENV PYTHONUNBUFFERED=1
ENV PYTHONDONTWRITEBYTECODE=1
ENV PIP_NO_CACHE_DIR=1
ENV PIP_DISABLE_PIP_VERSION_CHECK=1

# Creamos directorios estándar con permisos correctos
RUN mkdir -p /app /logs /data && \
    chown -R appuser:appgroup /app /logs /data

# Configurar pip para permitir instalaciones globales
ENV PIP_BREAK_SYSTEM_PACKAGES=1

# Esta imagen no se ejecuta sola, solo sirve como base.
# Por eso no tiene un comando CMD o ENTRYPOINT.

# Documentación de uso:
# FROM mi-empresa/base-alpine:1.0.0
# USER appuser
# WORKDIR /app
# ... resto de tu aplicación
