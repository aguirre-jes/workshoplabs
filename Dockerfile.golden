# Dockerfile usando Golden Image - Aplicación Flask API Status
# Este Dockerfile demuestra cómo usar una Golden Image corporativa como base

# ¡LA LÍNEA CLAVE!
# Ya no partimos de "alpine" o "python", sino de NUESTRA "Golden Image"
# Nota: Para demo local, usa "mi-empresa/base-alpine:1.0"
# En producción sería algo como: "acrworkshopcontainers2024.azurecr.io/mi-empresa/base-alpine:1.0"
FROM acrworkshopcontainers2024.azurecr.io/mi-empresa/base-alpine:1.0

# Información sobre esta aplicación específica
LABEL app.name="api-status"
LABEL app.version="1.0.0"
LABEL app.description="API de status usando Golden Image"

# Configuramos el directorio de trabajo (ya existe y tiene permisos correctos)
WORKDIR /app

# Cambiar al usuario estándar (ya viene configurado en la Golden Image)
# ¡No necesitamos crear el usuario, ya existe!
USER appuser

# Copiamos requirements.txt primero para aprovechar cache de Docker
# Ya no necesitamos instalar curl, jq, bash o python - ¡ya vienen en la Golden Image!
COPY --chown=appuser:appgroup requirements.txt .

# Instalamos dependencias Python
# pip ya está disponible en la Golden Image
RUN pip install --no-cache-dir -r requirements.txt

# Copiamos el código de la aplicación
COPY --chown=appuser:appgroup app.py .

# Exponemos el puerto (documentativo)
EXPOSE 8080

# Verificamos que todo está en orden (opcional, para debugging)
RUN python --version && \
    curl --version && \
    jq --version && \
    whoami

# Comando para ejecutar la aplicación
CMD ["python", "app.py"]

# Beneficios de usar Golden Image:
# ✅ Seguridad: Usuario sin privilegios pre-configurado
# ✅ Velocidad: Herramientas ya instaladas (curl, jq, python)
# ✅ Consistencia: Misma base para todos los proyectos
# ✅ Mantenimiento: Actualizaciones centralizadas en la Golden Image
# ✅ Compliance: Base auditada y aprobada por el equipo de seguridad
